### DISCUSSION TITLE
error running `mage test:unit` in main branch

### DISCUSSION THREAD

**Discussion by dangogh** [aquasecurity/trivy#7568] [Development]
First time contributor trying to run unit tests and I get the following error.  Could there be something incorrect in my setup?
```
mage test:unit
# github.com/aquasecurity/trivy/pkg/module/wasm
../../wasm/sdk.go:146:10: cannot use uintptr(size) (value of type uintptr) as int value in assignment
../../wasm/sdk.go:147:10: cannot use uintptr(size) (value of type uintptr) as int value in assignment
../../wasm/sdk.go:163:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
../../wasm/sdk.go:164:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
pkg/module/testdata/analyzer/analyzer.go:1: running "tinygo": exit status 1
Error: running "go generate pkg/module/testdata/analyzer/analyzer.go" failed with exit code 1
```
Running on a macbook pro M3 with Sonoma 14.6.1 and go1.22.4 darwin/arm64

**Comment by dangogh** [comment]
other details:
Above is on main branch with no changes.
tinygo version 0.33.0 darwin/arm64 (using go version go1.22.4 and LLVM version 18.1.2)

**Reply by knqyf263** [reply]
TinyGo might have added any breaking change. Did you try v0.31.1?
https://github.com/aquasecurity/trivy/blob/1f9fc13da4a1e7c76c978e4f8e119bfd61a0480e/aqua.yaml#L8

**Comment by dangogh** [comment]
You are correct.  I tried `v0.31.2` and `v0.32.0`.   The breaking change seems to be in `v0.32.0`.

I think the `uintptr` cast here should be removed -- it's unclear why it was there to start with:
https://github.com/aquasecurity/trivy/blob/37d549e5b86a1c5dce6710fbfd2310aec9abe949/pkg/module/wasm/sdk.go#L146-L147

**Reply by nikpivkin** [reply]
https://github.com/tinygo-org/tinygo/blob/c3032660c9e2e1fd5a188c6ac54902637a77b975/src/reflect/value.go#L743-L745

**Reply by knqyf263** [reply]
We just followed an example in wazero, but they changed to use unsafel.Slice. We can also migrate to it.
https://github.com/tetratelabs/wazero/pull/1436

**Reply by dangogh** [reply]
@knqyf263 would you like me to file an issue for that?   Otherwise, consider this discussion resolved.

**Reply by knqyf263** [reply]
OK. Closing. It would be great if you fix the issue with TinyGo v0.32.0.

**Reply by dangogh** [reply]
ok -- I'll start a discussion over there

**Reply by dangogh** [reply]
@knqyf263 
Actually, it appears to be an intentional known breaking change to make the reflect.*Header types compatible with the main Go implementation based on these discussions:

https://github.com/tinygo-org/tinygo/issues/4318
https://github.com/kelindar/binary/issues/18

**Reply by knqyf263** [reply]
OK, we want to fix our code, then.

**Reply by michaeljsaenz** [reply]
is there a temp workaround to get `mage test:unit` passing?

**Reply by knqyf263** [reply]
v0.31.2 should work.
https://github.com/aquasecurity/trivy/discussions/7568#discussioncomment-10732415
